### PR TITLE
Fix build warnings

### LINKED ISSUE #1611: [BUILD] Build warnings
There are multiple build warnings in the source code.

Example:

```
/home/malff/CODE/MARC_GITHUB/opentelemetry-cpp/api/include/opentelemetry/nostd/string_view.h:84:4: error: extra ‘;’ after in-class function definition [-Werror=extra-semi]
   84 |   };

```

### PR DESCRIPTION
**Pull request by marcalff**
Fixes #1611 

## Changes

Fix build warnings reported by gcc 10, with:
 -Wall -Werror
 -Wextra-semi
 -Werror=unused-parameter
 -Werror=undef

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [X] Changes in public API reviewed

### CODE REVIEW

File: exporters/otlp/test/otlp_http_metric_exporter_test.cc [hunk 1]
```diff
@@ -196,13 +196,15 @@ class OtlpHttpMetricExporterTestPeer : public ::testing::Test
     auto client                = mock_otlp_client.second;
     auto exporter = GetExporter(std::unique_ptr<OtlpHttpClient>{mock_otlp_http_client});
 
+#  if 0
```
Comment (ThomsonTan) [inline exporters/otlp/test/otlp_http_metric_exporter_test.cc:199]: Probably just remove the dead code?

Comment (marcalff) [reply exporters/otlp/test/otlp_http_metric_exporter_test.cc:199]: Given how metrics is still in development, not sure if this is:
- dead code (ok to remove then)
- not yet used code (which better stay)

so I took the safe choice.

File: sdk/src/metrics/state/temporal_metric_storage.cc [hunk 2]
```diff
@@ -69,7 +69,7 @@ bool TemporalMetricStorage::buildMetrics(CollectorHandle *collector,
                                                 instrument_descriptor_, aggregation_config_.get())
                                                 ->Merge(aggregation));
             merged_metrics->GetAllEnteries(
-                [](const MetricAttributes &attr, Aggregation &aggr) { return true; });
+                [](const MetricAttributes & /* attr */, Aggregation & /* aggr */) { return true; });
```
Comment (lalitb) [inline sdk/src/metrics/state/temporal_metric_storage.cc:72]: nit - could you remove this statement (line 71,72) to iterate over the merged entries, as it is not doing anything. Seems to be part of debugging, and I forgot to remove. Thanks :)

File: sdk/src/metrics/state/temporal_metric_storage.cc [hunk 3]
```diff
@@ -112,7 +112,7 @@ bool TemporalMetricStorage::buildMetrics(CollectorHandle *collector,
   else
   {
     merged_metrics->GetAllEnteries(
-        [](const MetricAttributes &attr, Aggregation &aggr) { return true; });
+        [](const MetricAttributes & /* attr */, Aggregation & /* aggr */) { return true; });
```
Comment (lalitb) [inline sdk/src/metrics/state/temporal_metric_storage.cc:115]: This one also can be removed - line 114, 115

File: exporters/otlp/test/otlp_http_metric_exporter_test.cc (hunk 1 above)
Comment (marcalff) [reply exporters/otlp/test/otlp_http_metric_exporter_test.cc:199]: @owent 

Please advise on drop / keep dead code, this comes from:

commit 24d5100ada0c6cb9ccadb5827e4889e657ed8644
Author: WenTao Ou <owentou@tencent.com>
Date:   Tue Jul 19 15:59:24 2022 +0800

**Review by lalitb — APPROVED** [review]
Looks good to me, but as it's bigger cleanup, will wait for one more approval :). Thanks for taking this.

File: sdk/src/metrics/state/temporal_metric_storage.cc (hunk 2 above)
Comment (marcalff) [reply sdk/src/metrics/state/temporal_metric_storage.cc]: Done

File: sdk/src/metrics/state/temporal_metric_storage.cc (hunk 3 above)
Comment (marcalff) [reply sdk/src/metrics/state/temporal_metric_storage.cc]: Done

File: exporters/otlp/test/otlp_http_metric_exporter_test.cc (hunk 1 above)
Comment (owent) [reply exporters/otlp/test/otlp_http_metric_exporter_test.cc:199]: > Given how metrics is still in development, not sure if this is:
> 
>     * dead code (ok to remove then)
> 
>     * not yet used code (which better stay)
> 
> 
> so I took the safe choice.

Yes, these codes can be removed. I forgot them and sorry for that.

**Review by owent — APPROVED** [review]
LGTM after the dead codes removed.

File: exporters/otlp/test/otlp_http_metric_exporter_test.cc (hunk 1 above)
Comment (marcalff) [reply exporters/otlp/test/otlp_http_metric_exporter_test.cc]: Thanks for confirming. Removed.

**Review by owent — APPROVED** [review]

**Comment by marcalff** [comment]
The DocFX check build failed due to this:

```
The install of docfx was NOT successful.
docfx not installed. An error occurred during installation:
 The remote server returned an error: (500) Internal Server Error. Internal Server Error
```

which looks like a transient error unrelated to the change.

@lalitb , @ThomsonTan 

This PR is ready for merge now.

**Comment by esigo** [comment]
related to #249

**Pull request merged**